app/controllers/posts_controller.rb

```diff
@@ -8,6 +8,22 @@ def show
     @upvote_comment = Comment.new
   end
 
+  def add
+  end
+
+  def autofill
+    @post = Post.new
+    begin
+      page = MetaInspector.new(link_params[:url])
+      @post.title = page.title
+      @post.description = page.best_description
+      @post.url = page.url
+    rescue MetaInspector::TimeoutError, MetaInspector::RequestError, MetaInspector::ParserError
+      redirect_to new_post_path and return
```

Comment:
can't we just reuse the `new` action for this all? We just use the metainspector if a url param is given. If not of if it fails we just render the plain view.
so we would not need an additional action.